src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -222,6 +227,49 @@ export class TabsTitleControl extends TitleControl {
 		});
 	}
 
+	private updateTabSizing(fromEvent: boolean): void {
+		const [tabsContainer, tabSizingFixedDisposables] = assertAllDefined(this.tabsContainer, this.tabSizingFixedDisposables);
+
+		tabSizingFixedDisposables.clear();
+
+		const options = this.accessor.partOptions;
+		if (options.tabSizing === 'fixed') {
+			tabsContainer.style.setProperty('--tab-sizing-fixed-max-width', `${options.tabSizingFixedMaxWidth}px`);
+
+			// For https://github.com/microsoft/vscode/issues/40290 we want to
+			// preserve the current tab widths as long as the mouse is over the
+			// tabs so that you can quickly close them via mouse click. For that
+			// we track mouse movements over the tabs container.
+
+			tabSizingFixedDisposables.add(addDisposableListener(tabsContainer, EventType.MOUSE_ENTER, () => {
+				this.isMouseOverTabs = true;
+			}));
+			tabSizingFixedDisposables.add(addDisposableListener(tabsContainer, EventType.MOUSE_LEAVE, () => {
+				this.isMouseOverTabs = false;
+				this.updateTabsFixedWith(false);
+			}));
+		} else if (fromEvent) {
+			tabsContainer.style.removeProperty('--tab-sizing-fixed-max-width');
+			this.updateTabsFixedWith(false);
+		}
+	}
+
+	private updateTabsFixedWith(fixed: boolean): void {
+		this.forEachTab((editor, index, tabContainer) => {
+			if (fixed) {
+				const { width } = tabContainer.getBoundingClientRect();
+
+				// Adjust width so the last tab doesn't wrap onto the next line due to tiny rounding errors
+				// introduced by fixed sizing just overflowing the tab bar's width.
+				const adjustedWidth = tabContainer.classList.contains('last-in-row') ? width - 0.1 : width;
```

Comment:
I am still not sure I understand how this can happen. Can we avoid rounding errors? This still seems like a hack to me, would be great to sort out.

Reply:
I don't think that we can get rid of the rounding errors. We get a measurement from `getBoundingBox` and when we set it as the `min-width`, the layout engine may get a slightly different number because the number is serialized and then parsed as a string.

I've created a codepen that illustrates the issue: https://codepen.io/jacekkopecky/pen/vYVvNqx?editors=0110

It looks like this: 
![2023-05-21 16 22 05](https://github.com/microsoft/vscode/assets/807315/1ee07e61-dd2e-434f-8762-cc7908791102)

If you don't see that in Chrome, try with a different zoom (ctrl/cmd-plus or minus); the problem doesn't seem to always show.

I have now committed a simpler solution where the tiny adjustment is done in CSS.

Reply:
Ok thanks!